provider/kubernetes/kubernetes.go

```diff
@@ -323,10 +333,46 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 				}
 			}
 		}
+
+		err = p.updateIngressStatus(i, k8sClient)
+		if err != nil {
+			log.Printf("cannot update Ingress %s/%s due to error: %s", i.Namespace, i.Name, err)
+		}
 	}
 	return &templateObjects, nil
 }
 
+func (p *Provider) updateIngressStatus(i *extensionsv1beta1.Ingress, k8sClient Client) error {
+	// Only process if an IngressEndpoint has been configured
+	if p.IngressEndpoint == nil {
+		return nil
+	}
+
+	if p.IngressEndpoint.PublishedService == "" {
+		return k8sClient.UpdateIngressStatus(i.Namespace, i.Name, p.IngressEndpoint.IP, p.IngressEndpoint.Hostname)
```

Comment:
Should we maybe emit a warning when one of the fields is an empty string? Or is that expected?

Reply:
I think is is expected that there would only be one of IP or Hostname ... perhaps we should check that they are not both empty?

Reply:
I added a check if the both are empty.